templates/terraform/nested_query.go.erb

```diff
@@ -159,9 +171,16 @@ func resource<%= resource_name -%>PatchDeleteEncoder(d *schema.ResourceData, met
   }
 
   updatedItems := append(currItems[:idx], currItems[idx+1:]...)
-  return map[string]interface{}{
+  res := map[string]interface{}{
     "<%= list_key %>": updatedItems,
-  }, nil
+  }
+  <% object.nested_query.keys.reverse[1..-1].each do |k| -%>
```

Comment:
we can also do a reduce here if you're interested in changing this output - I think the go would look nice but the ruby ERB less nice either way! It'd be like:
```
wrappedObj := <%= 
  object.nested_query.keys
            .reverse[1..-1]
            .reduce("res") { |res, k| "map[string]interface{\n\t#{k}:#{res},\n}" }
-%>
```
to print out:

```
wrappedObj := map[string]interface{
   "k0": map[string]interface{
      "k1": ...
         "k_last": res,
      },
   },
}
```

Reply:
Done, I opted for a bit of a compromise between the two. The one with reduce looks nice for this resource but it would need some extra handling for resources with only one key, which is the majority of them.
I also didn't bother checking the array length because it seems to succeed just fine if the array is too small (the slice of the array will just be empty).